WoWsShipBuilder.Web/Components/DispersionPlot.razor

```diff
@@ -0,0 +1,191 @@
+            ChartsHelper.EllipsePlanes.RealPlane => dispersionEllipse.VerticalRadius,
+            _ => dispersionEllipse.ProjectedOnPerpendicularToWaterVerticalRadius,
+        };
+
+        xRadius *= PlotScaling;
+        double yRadius = dispersionEllipse.HorizontalRadius * PlotScaling;
+
+        if (IsVertical)
+        {
+            (xRadius, yRadius) = (yRadius, xRadius);
+        }
+        return (xRadius, yRadius);
+    }
+    
+    private List<ChartsHelper.Point> GetHitPoints(DispersionEllipse dispersionEllipse)
+    {
+        List<(double x, double y)> hitPoints = EllipsePlane switch
+        {
+            ChartsHelper.EllipsePlanes.HorizontalPlane => dispersionEllipse.OnWaterHitPoints,
+            ChartsHelper.EllipsePlanes.RealPlane => dispersionEllipse.RealHitPoints,
+            _ => dispersionEllipse.PerpendicularToWaterHitPoints,
+        };
+
+        return IsVertical ? hitPoints.Select(point => new ChartsHelper.Point(point.x * PlotScaling, point.y * PlotScaling)).ToList() : hitPoints.Select(point => new ChartsHelper.Point(point.y * PlotScaling, point.x * PlotScaling)).ToList();
+    }
+    
+    private (double xRadius, double yRadius) GetInnerEllipse(DispersionEllipse dispersionEllipse)
+    {
+        double xRadius = EllipsePlane switch
+        {
+            ChartsHelper.EllipsePlanes.HorizontalPlane => dispersionEllipse.ProjectedOnWaterVerticalRadiusHalfHitPoints,
+            ChartsHelper.EllipsePlanes.RealPlane => dispersionEllipse.VerticalRadiusHalfHitPoints,
+            _ => dispersionEllipse.ProjectedOnPerpendicularToWaterVerticalRadiusHalfHitPoints,
+        };
+
+        xRadius *= PlotScaling;
+        double yRadius = dispersionEllipse.HorizontalRadiusHalfHitPoints * PlotScaling;
+
+        if (IsVertical)
+        {
+            (xRadius, yRadius) = (yRadius, xRadius);
+        }
+        return (xRadius, yRadius);
+    }
+    
+    private Size GetFusoReference()
+    {
+        if (FusoPosition == FusoPositions.DontShow)
+        {
+            return new(0, 0);
+        }
+        
+        const double length = 212.7;
+        const double width = 32.2;
+        const int height = 45;
+
+        double yRadius = width;
+        double xRadius = length;
+
+        if (FusoPosition == FusoPositions.Broadside)
+        {
+            xRadius = width;
+            yRadius = length;
+        }
+
+        if (EllipsePlane == ChartsHelper.EllipsePlanes.VerticalPlane)
+        {
+            xRadius = height;
+        }
+
+        xRadius *= PlotScaling;
+        yRadius *= PlotScaling;
+
+        if (IsVertical)
+        {
+            (xRadius, yRadius) = (yRadius, xRadius);
+        }
+        return new(xRadius, yRadius);
+    }
+    
+    public sealed record DrawingData(double XRadiusOuterEllipse, double YRadiusOuterEllipse, double XRadiusInnerEllipse, double YRadiusInnerEllipse, List<ChartsHelper.Point> HitPoints);
```

Comment:
Extract records into separate file if they have to be public.
Instead, consider making them private and use collocated JS with interop being placed in this component